pkg/dependenciesdistributor/dependencies_distributor_test.go

```diff
@@ -41,6 +45,194 @@ import (
+						Name: "foo",
+					},
+				},
+			},
+			wantQueueSize: 2,
+		},
+		{
+			name: "do not update the object, no specification changed",
+			args: args{
+				oldObj: &corev1.Node{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Node",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "bar",
+					},
+				},
+				newObj: &corev1.Node{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Node",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "bar",
+					},
+				},
+			},
+			wantQueueSize: 0,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockWorker := &MockAsyncWorker{}
+			d := &DependenciesDistributor{
+				resourceProcessor: mockWorker,
+			}
+
+			d.OnUpdate(tt.args.oldObj, tt.args.newObj)
+
+			gotQueueSize := len(mockWorker.GetQueue())
+			if gotQueueSize != tt.wantQueueSize {
+				t.Errorf("OnUpdate() want queue size %v, got %v", tt.wantQueueSize, gotQueueSize)
+			}
+		})
+	}
+}
+
+func Test_reconcileResourceTemplate(t *testing.T) {
+	type args struct {
+		key util.QueueKey
+	}
+	type fields struct {
+		Client client.Client
+	}
+	tests := []struct {
+		name    string
+		args    args
+		fields  fields
+		wantErr bool
+	}{
+		{
+			name: "reconcile resource template",
+			args: args{
+				key: &LabelsKey{
+					ClusterWideKey: keys.ClusterWideKey{
+						Group:     "apps",
+						Version:   "v1",
+						Kind:      "Deployment",
+						Name:      "demo-app",
+						Namespace: "test",
+					},
+					Labels: map[string]string{
+						"app": "test",
+					},
+				},
+			},
+			fields: fields{
+				Client: func() client.Client {
+					Scheme := runtime.NewScheme()
+					utilruntime.Must(scheme.AddToScheme(Scheme))
+					utilruntime.Must(workv1alpha2.Install(Scheme))
+					rb := &workv1alpha2.ResourceBinding{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:            "test-binding",
+							Namespace:       "test",
+							ResourceVersion: "1000",
+							Labels: map[string]string{
+								"app": "test",
+							},
+							Annotations: map[string]string{
+								dependenciesAnnotationKey: "[{\"apiVersion\":\"apps/v1\",\"kind\":\"Deployment\",\"namespace\":\"test\",\"name\":\"demo-app\"}]",
+							},
+						},
+						Spec: workv1alpha2.ResourceBindingSpec{
+							Resource: workv1alpha2.ObjectReference{
+								APIVersion:      "apps/v1",
+								Kind:            "Deployment",
+								Namespace:       "test",
+								Name:            "demo-app",
+								ResourceVersion: "22222",
+							},
+						},
+					}
+					return fake.NewClientBuilder().WithScheme(Scheme).WithObjects(rb).Build()
+				}(),
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			d := &DependenciesDistributor{
+				Client:       tt.fields.Client,
+				genericEvent: make(chan event.TypedGenericEvent[*workv1alpha2.ResourceBinding], 1),
+			}
+
+			err := d.reconcileResourceTemplate(tt.args.key)
```

Comment:
In addition to determining err, is it also possible to determine the expectations of related resources? For example, the length of genericEvent.